File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        
        double a = Math.sqrt(n);
        long b = (long) a;
        
        if(a - b != 0){
            answer = -1;
        }else {
            answer = (b + 1) * (b + 1);
            //answer = (long) Math.pow(a+1, 2);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

long long solution(long long n) {
    long long answer = 0;
    long long t = sqrt(n);
        
    if(t * t != n){
        answer = -1;
    }else {
        answer = (t + 1) * (t + 1);
    }
    
    return answer;
}